### PR TITLE
Documentation: upgrade grunt-jsdoc and fix invalid jsdoc comments

### PR DESCRIPTION
**Pull request by nemtsov**

### CODE REVIEW

File: Controller/Dialog/Modal.js [hunk 1]
```diff
@@ -20,7 +20,8 @@ define([
     },
 
     /**
-     * @abstract Function expected to generate a Promise
+     * Function expected to generate a Promise
+     * @abstract
```
Comment (joesepi) [inline Controller/Dialog/Modal.js]: Why this change? Just curious.

**Comment by joesepi** [comment]
👍🏼

File: Controller/Dialog/Modal.js (hunk 1 above)
Comment (nemtsov) [inline Controller/Dialog/Modal.js]: @joesepi jsdoc doesn't allow anything after the `@abstract` keyword. (http://usejsdoc.org/tags-abstract.html)

Comment (joesepi) [inline Controller/Dialog/Modal.js]: Cool. Thanks!

**Comment by kaicataldo** [comment]
:+1:
